pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -331,13 +342,18 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 			withMessage(None, "").
 			withRunningPhase(),
 	)
-
-	annotations = map[string]string{
-		lastSuccessfulConfiguration: string(mdb.Spec.ToJson()),
-	}
-	r.setAnnotations(mdb.NamespacedName(), annotations)
-
-	if err != nil {
+	if err == nil {
+		// If we could get all the way up to here, then save the current spec as the last
+		// successfully applied.
+		lastSuccessfulConfigurationSaved, err := mdb.Spec.ToJson()
+		if err != nil {
+			r.log.Error("Error converting Spec to Json: %s", err)
```

Comment:
We should exit early here, otherwise we are going to attempt to add a junk value as an annotation!